ios/FluentUI/Vnext/AvatarGroup/AvatarGroup.swift

```diff
@@ -0,0 +1,202 @@
+    }
+
+    func getAvatarState(_ atIndex: Int) -> MSFAvatarState? {
+        guard atIndex < avatars.count else {
+            return nil
+        }
+        return avatars[atIndex]
+    }
+
+    func deleteAvatar(atIndex: Int) {
+        guard atIndex < avatars.count else {
+            return
+        }
+        avatars.remove(at: atIndex)
+    }
+
+    @Published var avatars: [MSFAvatarStateImpl] = []
+    @Published var maxDisplayedAvatars: Int = Int.max
+    @Published var overflowCount: Int = 0
+
+    var size: MSFAvatarSize {
+        get {
+            return tokens.size
+        }
+        set {
+            tokens.size = newValue
+        }
+    }
+
+    var style: MSFAvatarGroupStyle {
+        get {
+            return tokens.style
+        }
+        set {
+            tokens.style = newValue
+        }
+    }
+
+    var tokens: MSFAvatarGroupTokens
+
+    init(style: MSFAvatarGroupStyle,
+         size: MSFAvatarSize) {
+        self.tokens = MSFAvatarGroupTokens(style: style,
+                                           size: size)
+        super.init()
+    }
+}
+
+/// View that represents the AvatarGroup
+public struct AvatarGroup: View {
+    @Environment(\.theme) var theme: FluentUIStyle
+    @Environment(\.windowProvider) var windowProvider: FluentUIWindowProvider?
+    @ObservedObject var state: MSFAvatarGroupStateImpl
+    @ObservedObject var tokens: MSFAvatarGroupTokens
+
+    init(style: MSFAvatarGroupStyle, size: MSFAvatarSize) {
+        let state = MSFAvatarGroupStateImpl(style: style, size: size)
+        self.state = state
+        self.tokens = state.tokens
+    }
+
+    public var body: some View {
+        let avatars: [MSFAvatarStateImpl] = state.avatars
+        let maxDisplayedAvatars: Int = avatars.prefix(state.maxDisplayedAvatars).count
+        let overflowCount: Int = (avatars.count > maxDisplayedAvatars ? avatars.count - maxDisplayedAvatars : 0) + state.overflowCount
+        let interspace: CGFloat = tokens.interspace
+        let ringOuterGap: CGFloat = tokens.ringOuterGap
+        let ringOffset: CGFloat = tokens.ringThickness + tokens.ringInnerGap + ringOuterGap
+        let size: CGFloat = tokens.size.size
+        let x: CGFloat = size + tokens.interspace - tokens.ringThickness
+        HStack(spacing: 0) {
+            ForEach(0 ..< maxDisplayedAvatars, id: \.self) { index in
+                // If the avatar is part of Stack style and is not the last avatar in the sequence, create a cutout
+                let needsCutout = tokens.style == .stack && (overflowCount > 0 || index + 1 < maxDisplayedAvatars)
+                let currentRingCheck = avatars[index].isRingVisible
+                let nextRingCheck = index + 1 < maxDisplayedAvatars ? avatars[index + 1].isRingVisible : false
+                AvatarView(avatars[index])
+                    .modifyIf(needsCutout, { view in
+                        view.mask(AvatarCutout(xOrigin: currentRingCheck ? x + ringOffset : x,
```

Comment:
Can you store the results of these nested ternary operators into constants with descriptive names?
It will greatly improve the code readability.
The one on line 127 is very hard to read:

```
 .padding(.trailing, tokens.style == .stack ?
                                (currentRingCheck ? (nextRingCheck ? interspace - (ringOffset + ringOuterGap) : interspace - ringOffset) :
                                    (nextRingCheck ? interspace - ringOuterGap : interspace)) : interspace)
```

Reply:
How about creating constants for xOrigin, yOrigin and cutoutSize?